packages/connector-local/src/connector.ts

```diff
@@ -365,17 +372,28 @@ export default class LocalConnector implements IConnector {
         };
     }
 
-    public async collect(target: url.URL, options?: IFetchOptions) {
+    private getTargetString(target: URL, isFile: boolean) {
+        const href = target.href;
+
+        if (isFile) {
+            return href;
+        }
+
+        return href[href.length - 1] === '/' ? href : `${href}/`;
+    }
+
+    public async collect(target: URL, options?: IFetchOptions) {
         /** The target in string format */
-        const href: string = this._href = target.href;
+        const pathString = asPathString(target);
+        const file = isFile(pathString);
+        const href = this._href = this.getTargetString(target, file);
```

Comment:
What does this fix? And do we have a test that covers it?

Reply:
This was to get the right path/URL when we were analyzing a folder, instead of a file. I think this is not going to be necessary anymore.